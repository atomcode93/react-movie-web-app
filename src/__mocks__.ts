export const initialState = {
  auth: {
    error: false,
    isAuthenticated: false,
    user: null,
  },
  favorites: {
    error: false,
    favoritedIds: [],
    favoritedMovies: null,
  },
  filters: {
    order: {
      label: 'Descending',
      value: 'desc',
    },
    rating: {
      max: 10,
      min: 5,
    },
    runtime: {
      max: 250,
      min: 45,
    },
    sort_by: {
      label: 'Rating',
      value: 'vote_average',
    },
    year: 2019,
  },
  modals: {
    args: null,
    open: false,
    type: null,
  },
  movies: {
    error: false,
    favorites: [],
    loading: false,
    movie: null,
    movies: null,
    popularMovies: null,
    soonMovies: null,
    topRatedMovies: null,
    watchLater: null,
  },
  router: {
    action: 'POP',
    location: {
      hash: '',
      pathname: '/',
      search: '',
      state: undefined,
    },
  },
  watchLater: {
    error: false,
    watchLaterIds: [],
    watchLaterMovies: null,
  },
};
