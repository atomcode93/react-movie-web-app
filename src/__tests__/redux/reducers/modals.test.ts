import {
  actions as modalsActions,
  reducer as modalsReducers,
} from 'redux/ducks/modals.duck';
import { initialState } from '__mocks__';

const modalsInitialState = initialState.modals;

describe('Modals Reducer', () => {
  test('should setup default modals state', () => {
    const state = modalsReducers(undefined, { type: '@@INIT' });
    expect(state).toEqual(modalsInitialState);
  });
  test('should open modal', () => {
    const action = modalsActions.openModal({
      type: 'video',
    });
    const state = modalsReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
  test('should close modal', () => {
    const action = modalsActions.closeModal();
    const state = modalsReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
