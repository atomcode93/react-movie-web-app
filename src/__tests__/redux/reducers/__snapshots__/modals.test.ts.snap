// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals Reducer should close modal 1`] = `
Object {
  "args": null,
  "open": false,
  "type": null,
}
`;

exports[`Modals Reducer should open modal 1`] = `
Object {
  "args": null,
  "open": true,
  "type": "video",
}
`;
